platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/RuntimeTypeAdapterFactory.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2011 Google Inc.
```

Comment:
🤔 Is it alright to do that? It seems like a mix of multiple sources

Reply:
I've read the license agreement and  iiuc the requirement is i) correct attribution in the header, ii) a link to the sources, and iii) an entry in the `NOTICE` file. I'll look into implementing a replacement for `RuntimeTypeAdapterFactory.java`. If I find that we still need this class then I'll double check the legal requirement.

Reply:
I was able to get rid of this class.